polyply/src/build_system.py

```diff
@@ -0,0 +1,164 @@
+        for node in molecule.nodes:
+            if "position" in molecule.nodes:
+                positions[idx, :] = molecule.nodes["position"]
+
+            resname = molecule.nodes[node]["resname"]
+            atom_types.append(resname)
+            nodes_to_gndx[(mol_count, node)] = idx
+            idx += 1
+        mol_count += 1
+    inter_matrix = {}
+    for res_A, res_B in itertools.combinations(set(atom_types), r=2):
+        inter_matrix[frozenset([res_A, res_B])] = lorentz_berthelot_rule(topology.volumes[res_A],
+                                                                       topology.volumes[res_B], 1, 1)
+    for resname, vdw_radii in topology.molecules[0].volumes.items():
+        inter_matrix[frozenset([resname, resname])] = vdw_radii
+
+    return positions, atom_types, nodes_to_gndx, inter_matrix
+
+
+class BuildSystem():
+    """
+    Compose a system of molecules according
+    to the definitions in the topology file.
+    """
+
+    def __init__(self, density, n_grid_points=250, maxiter=20, box_size=None):
+        self.density = density
+        self.n_grid_points = n_grid_points
+        self.maxiter = maxiter
+        self.box_size = box_size
+
+    def _handle_random_walk(self, molecule, topology, positions, inter_matrix,
+                            nodes_to_gndx, atom_types, mol_idx, vector_sphere):
+        step_count = 0
+        while True:
+            start = self.box_grid[np.random.randint(len(self.box_grid), size=3)]
+            processor = RandomWalk(positions,
+                                   nodes_to_gndx,
+                                   atom_types,
+                                   inter_matrix,
+                                   start=start,
+                                   mol_idx=mol_idx,
+                                   topology=topology,
+                                   maxiter=50,
+                                   maxdim=self.maxdim,
+                                   vector_sphere=vector_sphere)
+
+            positions, processor.run_molecule(molecule)
+            if processor.success:
+                return True, processor.positions
+            elif step_count == self.maxiter:
+                return False, processor.positions
+            else:
+                step_count += 1
+
+    def _compose_system(self, topology):
+        """
+        Place the molecules of the system into a box
+        and optimize positions to meet density.
+
+        Parameters
+        ----------
+        topology:  :class:`vermouth.system`
+        density: foat
+           density of the system in kg/cm3
+
+        Returns
+        --------
+        system
+        """
+        self.box_size = round(_compute_box_size(topology, self.density),5)
+        self.box_grid = np.arange(0, self.box_size, self.box_size/self.n_grid_points)
+        self.maxdim = np.array([self.box_size, self.box_size, self.box_size])
+        topology.box = (self.box_size, self.box_size, self.box_size)
```

Comment:
Impolite to set other people's object attributes. But ok.

Reply:
yes, this was a quick fix basically